gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebmasterDataFetcherImpl.java

```diff
@@ -0,0 +1,383 @@
+   *
+   */
+  @Override
+  public Collection<ProducerJob> getAllPages(String startDate, String endDate, String country, int rowLimit)
+      throws IOException {
+    if (!_jobs.isEmpty()) {
+      LOG.info("Service got hot started.");
+      return _jobs;
+    }
+
+    ApiDimensionFilter countryFilter = GoogleWebmasterFilter.countryEqFilter(country);
+
+    List<GoogleWebmasterFilter.Dimension> requestedDimensions = new ArrayList<>();
+    requestedDimensions.add(GoogleWebmasterFilter.Dimension.PAGE);
+
+    Collection<String> allPages =
+        _client.getPages(_siteProperty, startDate, endDate, country, rowLimit, requestedDimensions,
+            Arrays.asList(countryFilter), 0);
+    int actualSize = allPages.size();
+
+    if (rowLimit < GoogleWebmasterClient.API_ROW_LIMIT || actualSize < GoogleWebmasterClient.API_ROW_LIMIT) {
+      LOG.info(String.format("A total of %d pages fetched for property %s at country-%s from %s to %s", actualSize,
+          _siteProperty, country, startDate, endDate));
+    } else {
+      int expectedSize = getPagesSize(startDate, endDate, country, requestedDimensions, Arrays.asList(countryFilter));
+      LOG.info(String.format("Total number of pages is %d for market-%s from %s to %s", expectedSize,
+          GoogleWebmasterFilter.countryFilterToString(countryFilter), startDate, endDate));
+      Queue<Pair<String, FilterOperator>> jobs = new ArrayDeque<>();
+      expandJobs(jobs, _siteProperty);
+
+      allPages = getPages(startDate, endDate, requestedDimensions, countryFilter, jobs);
+      allPages.add(_siteProperty);
+      actualSize = allPages.size();
+      if (actualSize != expectedSize) {
+        LOG.warn(
+            String.format("Expected page size for country-%s is %d, but only able to get %d", country, expectedSize,
+                actualSize));
+      }
+      LOG.info(String.format("A total of %d pages fetched for property %s at country-%s from %s to %s", actualSize,
+          _siteProperty, country, startDate, endDate));
+    }
+
+    ArrayDeque<ProducerJob> jobs = new ArrayDeque<>(actualSize);
+    for (String page : allPages) {
+      jobs.add(new SimpleProducerJob(page, startDate, endDate));
+    }
+    return jobs;
+  }
+
+  private int getPagesSize(final String startDate, final String endDate, final String country,
+      final List<Dimension> requestedDimensions, final List<ApiDimensionFilter> apiDimensionFilters)
+      throws IOException {
+    int REQUESTS_COUNT_EACH_ROUND = 4;
+    //Each round will give you 20K(4*5000) pages. Doing 100 rounds can give 2 million pages.
+    //Normally 1 week has less than 200K pages. So 2 million pages are more than enough.
+    int MAXIMUM_ROUNDS = 100;
```

Comment:
Can we please make requests_count_each_round and maximum_rounds configurable, so that we can tweak velocity and request counts per round - should anything change (quota / network / google's API limitations) without re-deployment